### DISCUSSION TITLE
Chat

### DISCUSSION THREAD

**Discussion by Raspberry-Pi-Reviews** [Raspberry-Pi-Reviews/Suda-Music#5] [General]
hi

**Comment by Crilum** [comment]
I'm going to create a new PR, cause my current one is unmergeable..

**Reply by Raspberry-Pi-Reviews** [reply]
im just gonna do this and write a reply from now on lol

**Comment by Crilum** [comment]
Got another PR up.

**Reply by Raspberry-Pi-Reviews** [reply]
ok

**Reply by Raspberry-Pi-Reviews** [reply]
done, is there a way to set this discussion so only me and you could see and type in it?

**Reply by Crilum** [reply]
Not sure, but it'd probably be in the Settings tab

**Reply by Raspberry-Pi-Reviews** [reply]
ok ill look in there

**Reply by Raspberry-Pi-Reviews** [reply]
do you want to work on this app together? cuz I can add you as a collaborator

**Reply by Crilum** [reply]
Yeah, sure!

**Reply by Raspberry-Pi-Reviews** [reply]
ok!

**Reply by Raspberry-Pi-Reviews** [reply]
I sent an invite

**Reply by Raspberry-Pi-Reviews** [reply]
im gonna head to bed, its 11:32 where im at night. ill probably work on the app tomorrow, also if you go to my profile I have unfinished repositories but theres other apps.

**Reply by Crilum** [reply]
I accepted. Thanks!

**Reply by Raspberry-Pi-Reviews** [reply]
no problem :)

**Reply by Raspberry-Pi-Reviews** [reply]
do you have discord?

**Reply by Crilum** [reply]
No, sorry.. :man_shrugging:

**Comment by Crilum** [comment]
What do you think about creating a simple bash script to be able to just run `suda-music` in the Terminal? 

Here's a simple script:
```bash
#!/bin/bash

python $HOME/Suda-Music
```

We could also make it so it detects if python is installed, and installs it if it isn't, or has flags (like to preload a song)

**Reply by Raspberry-Pi-Reviews** [reply]
I like it

**Reply by Raspberry-Pi-Reviews** [reply]
im gonna try and make a volume slider instead of the volume buttons

**Reply by Crilum** [reply]
A more complicated (asks user to install Python if the script couldn't find it) version:

```bash
#!/bin/bash

if [ ! command -v python &>/dev/null ] && [ ! command -v python2 &>/dev/null ] && [ ! command -v python3 &>/dev/n>
read -t 30 -p 'Python is required to run Suda Music, but is not installed! Would you like to install it? Timeout >
if [ "$?" == "142" ]; then
echo -e '\e[0;31mInput timed out... Exiting.'; exit 1
else
case $input in
        y|Y|Yes|YES|yes|"") sudo apt install python3 ;;
        n|N|No|NO|no) echo -e "\e[0;31m\nUser error: user chose not to install Python!\e[0m Exiting..."; exit 1 ;;
esac
fi
fi

python $HOME/Suda-Music/musicplayer.py
```

**Reply by Crilum** [reply]
>im gonna try and make a volume slider instead of the volume buttons

A slider would definitely be better!

**Reply by Crilum** [reply]
Can I change the install script to create the above script?

**Reply by Raspberry-Pi-Reviews** [reply]
yeah sure go ahead

**Reply by Crilum** [reply]
check out cdac0f6

**Reply by Raspberry-Pi-Reviews** [reply]
so I cant figure out how to make a slider rn so im just gonna have one button so like if you click it 10 times that volume is max and if you click it 1 time, the volume is min

**Reply by Raspberry-Pi-Reviews** [reply]
also could you make the README.md file better looking?

**Comment by Raspberry-Pi-Reviews** [comment]
How does this look?
![image](https://user-images.githubusercontent.com/74014196/151684649-37c4251d-59d4-40b8-b074-0afec3889538.png)

**Reply by Crilum** [reply]
Um, well, no offense intended, but I don't think one volume button is better than two.. couldn't you make an `if` statement like this:

```python
def volumeup(self):
        while True:
               if mixer.music.volume(0.9) # This probably isn't the right syntax(?) for this library
                       break
                else
                       self.bttn_clicks1 += 1
#continue with the rest of the function

```
and another one for `volumedown`.

**Reply by Raspberry-Pi-Reviews** [reply]
genius lmfao

**Reply by Raspberry-Pi-Reviews** [reply]
ill get working on that right now lol

**Reply by Raspberry-Pi-Reviews** [reply]
so a little problem lol, when I tried to do that I pressed the volume button and the app froze

**Reply by Crilum** [reply]
mmm... could you upload the code? I'll see if I can troubleshoot it..

**Reply by Raspberry-Pi-Reviews** [reply]
yeah

**Comment by Raspberry-Pi-Reviews** [comment]
from tkinter import *
from tkinter import filedialog
from pygame import mixer
from tkinter import ttk
import time
class MusicPlayer:
    def __init__(self, window ):
        window.geometry('650x650'); window.title('Suda Music'); window.resizable(0,0); window.configure(background = "lightblue")
        self.bttn_clicks = 0
        self.bttn_clicks1 = 0
        self.background = "white"
        buttonframe = LabelFrame(root,text="Control Panel",font=("times new roman",15,"bold"),bg="grey",fg="white",bd=7,relief=GROOVE)
        sliderframe = LabelFrame(root,text="Volume Control",font=("times new roman",15,"bold"),bg="grey",fg="white",bd=7,relief=GROOVE)
        infoframe = LabelFrame(root,text="Updates",font=("times new roman",15,"bold"),bg="grey",fg="white",bd=7,relief=GROOVE)
        
        Load = Button(window, text = 'Load',  width = 10, height = 2, font = ('Times', 10, "bold"), bg="grey", fg="white", bd=4, relief=GROOVE, command = self.load)
        Play = Button(window, text = 'Play',  width = 10, height = 2, font = ('Times', 10 , "bold"), bg="grey", fg="white", bd=4, relief=GROOVE, command = self.play)
        volumeupButton = Button(window, text = 'Volume',  width = 10, height = 2, font = ('Times', 10, "bold"), bg="grey", fg="white", bd=4, relief=GROOVE, command = self.volumeup)
        Pause = Button(window,text = 'Pause',  width = 10, height = 2, font = ('Times', 10, "bold"), bg="grey", fg="white", bd=4, relief=GROOVE, command = self.pause)
        Stop = Button(window ,text = 'Stop',  width = 10, height = 2, font = ('Times', 10, "bold"), bg="grey", fg="white", bd=4, relief=GROOVE, command = self.stop)
        label = Label(root, text = "Suda Music (v1.01)", bd=5, bg = "black", fg = "white", font =("times new roman",15,"bold"),relief=GROOVE)
        label.pack()
        label2 = Label(root, text = "- Click the volume button 10 times for max volume", bg = "grey", fg = "white", font =("times new roman",15,"bold"))
        label2.pack()
        label4 = Label(root, text = "- Changed Background Color From Light Grey To Light Blue", bg = "grey", fg = "white", font =("times new roman",15,"bold"))
        label4.pack()
        label3 = Label(root, text = "- Added New Control Panel Design", bg = "grey", fg = "white", font =("times new roman",15,"bold"))
        label3.pack()
        Load.place(x=45,y=80);Play.place(x=200,y=80);Pause.place(x=355,y=80);Stop.place(x=505,y=80);buttonframe.place(x=25,y=50,width=600,height=100);label.place(x=235,y=8)
        sliderframe.place(x=25,y=170,width=170,height=100)
       
        volumeupButton.place(x=55, y=200)
        infoframe.place(x=25,y=300,width=600,height=300)
        label2.place(x=45,y=330)
        label3.place(x=45,y=360)
        label4.place(x=45,y=390)
        
       
        
        self.music_file = False
        self.playing_state = False
    def load(self):
        self.music_file = filedialog.askopenfilename()
    def play(self):
        if self.music_file:
            mixer.init()
            mixer.music.load(self.music_file)
            mixer.music.play()
    def pause(self):
        if not self.playing_state:
            mixer.init()
            mixer.music.pause()
            self.playing_state=True
        else:
            mixer.music.unpause()
            self.playing_state = False
    def stop(self):
        mixer.init()
        mixer.music.stop()
        
    
        
    def volumeup(self):
        while True:
               mixer.init()
               if mixer.music.set_volume(0.9):
                  
                  break
               else:
                       self.bttn_clicks1 += 1
            #print("Total Clicks: " + str(self.bttn_clicks))
    
    def volumedown(self):
        self.bttn_clicks1 += 1
        while True:  
            if self.bttn_clicks1==1:
                mixer.init()
                mixer.music.set_volume(0.1)
            elif self.bttn_clicks1==2:
                mixer.init()
                mixer.music.set_volume(0.2)
            elif self.bttn_clicks1==3:
                mixer.init()
                mixer.music.set_volume(0.3)
            elif self.bttn_clicks1==4:
                mixer.init()
                mixer.music.set_volume(0.4)
            elif self.bttn_clicks1==5:
                mixer.init()
                mixer.music.set_volume(0.5) 
            elif self.bttn_clicks1==6:
                mixer.init()
                mixer.music.set_volume(0.6)
            elif self.bttn_clicks1==7:
                mixer.init()
                mixer.music.set_volume(0.7)
            elif self.bttn_clicks1==8:
                mixer.init()
                mixer.music.set_volume(0.8)
            elif self.bttn_clicks1==9:
                mixer.init()
                mixer.music.set_volume(0.9)
            elif self.bttn_clicks1==10:
                mixer.init()
                mixer.music.set_volume(1.0)
                
                self.bttn_clicks1=0 
            #print("Total Clicks: " + str(self.bttn_clicks))
            break
                        
                
root = Tk()
app= MusicPlayer(root)
root.mainloop()

**Reply by Raspberry-Pi-Reviews** [reply]
all of that is the code it was being dumb

**Comment by Raspberry-Pi-Reviews** [comment]
did you figure it out?

**Comment by Raspberry-Pi-Reviews** [comment]
I kept trying but I got no where

**Comment by Crilum** [comment]
This is sort of what I meant for `volumeup()`, but I get error: 
```
Traceback (most recent call last):
  File "/home/pi/Suda-Music/tryvolup.py", line 139, in <module>
    app = MusicPlayer(root)
TypeError: MusicPlayer() takes no arguments
```
Code:
```python
from tkinter import *
from tkinter import filedialog
from pygame import mixer
from tkinter import ttk
import time
class MusicPlayer:
    def init(self, window ):
        window.geometry('650x650'); window.title('Suda Music'); window.resizable(0,0); window.configure(background = "lightblue")
        self.bttn_clicks = 0
        self.bttn_clicks1 = 0
        self.background = "white"
        buttonframe = LabelFrame(root,text="Control Panel",font=("times new roman",15,"bold"),bg="grey",fg="white",bd=7,relief=GROOVE)
        sliderframe = LabelFrame(root,text="Volume Control",font=("times new roman",15,"bold"),bg="grey",fg="white",bd=7,relief=GROOVE)
        infoframe = LabelFrame(root,text="Updates",font=("times new roman",15,"bold"),bg="grey",fg="white",bd=7,relief=GROOVE)
        Load = Button(window, text = 'Load',  width = 10, height = 2, font = ('Times', 10, "bold"), bg="grey", fg="white", bd=4, relief=GROOVE, command = self.load)
        Play = Button(window, text = 'Play',  width = 10, height = 2, font = ('Times', 10 , "bold"), bg="grey", fg="white", bd=4, relief=GROOVE, command = self.play)
        volumeupButton = Button(window, text = 'Volume',  width = 10, height = 2, font = ('Times', 10, "bold"), bg="grey", fg="white", bd=4, relief=GROOVE, command = self.volumeup)
        Pause = Button(window,text = 'Pause',  width = 10, height = 2, font = ('Times', 10, "bold"), bg="grey", fg="white", bd=4, relief=GROOVE, command = self.pause)
        Stop = Button(window ,text = 'Stop',  width = 10, height = 2, font = ('Times', 10, "bold"), bg="grey", fg="white", bd=4, relief=GROOVE, command = self.stop)
        label = Label(root, text = "Suda Music (v1.01)", bd=5, bg = "black", fg = "white", font =("times new roman",15,"bold"),relief=GROOVE)
        label.pack()
        label2 = Label(root, text = "- Click the volume button 10 times for max volume", bg = "grey", fg = "white", font =("times new roman",15,"bold"))
        label2.pack()
        label4 = Label(root, text = "- Changed Background Color From Light Grey To Light Blue", bg = "grey", fg = "white", font =("times new roman",15,"bold"))
        label4.pack()
        label3 = Label(root, text = "- Added New Control Panel Design", bg = "grey", fg = "white", font =("times new roman",15,"bold"))
        label3.pack()
        Load.place(x=45,y=80);Play.place(x=200,y=80);Pause.place(x=355,y=80);Stop.place(x=505,y=80);buttonframe.place(x=25,y=50,width=600,height=100);label.place(x=235,y=8)
        sliderframe.place(x=25,y=170,width=170,height=100)
        volumeupButton.place(x=55, y=200)
        infoframe.place(x=25,y=300,width=600,height=300)
        label2.place(x=45,y=330)
        label3.place(x=45,y=360)
        label4.place(x=45,y=390)
        self.music_file = False
        self.playing_state = False
def load(self):
    self.music_file = filedialog.askopenfilename()
def play(self):
    if self.music_file:
        mixer.init()
        mixer.music.load(self.music_file)
        mixer.music.play()
def pause(self):
    if not self.playing_state:
        mixer.init()
        mixer.music.pause()
        self.playing_state=True
    else:
        mixer.music.unpause()
        self.playing_state = False
def stop(self):
    mixer.init()
    mixer.music.stop()
    

    
def volumeup(self):
    while True:
        mixer.init()
        if mixer.music.get_volume() == '1.0':
            exit
        else:
            self.bttn_clicks1 += 1
            if self.bttn_clicks1==1:
                mixer.init()
                mixer.music.set_volume(0.1)
            elif self.bttn_clicks1==2:
                mixer.init()
                mixer.music.set_volume(0.2)
            elif self.bttn_clicks1==3:
                mixer.init()
                mixer.music.set_volume(0.3)
            elif self.bttn_clicks1==4:
                mixer.init()
                mixer.music.set_volume(0.4)
            elif self.bttn_clicks1==5:
                mixer.init()
                mixer.music.set_volume(0.5) 
            elif self.bttn_clicks1==6:
                mixer.init()
                mixer.music.set_volume(0.6)
            elif self.bttn_clicks1==7:
                mixer.init()
                mixer.music.set_volume(0.7)
            elif self.bttn_clicks1==8:
                mixer.init()
                mixer.music.set_volume(0.8)
            elif self.bttn_clicks1==9:
                mixer.init()
                mixer.music.set_volume(0.9)
            elif self.bttn_clicks1==10:
                mixer.init()
                mixer.music.set_volume(1.0)
        
            self.bttn_clicks1=0 
            break
            #print("Total Clicks: " + str(self.bttn_clicks))

def volumedown(self):
    self.bttn_clicks1 += 1
    while True:  
        if self.bttn_clicks1==1:
            mixer.init()
            mixer.music.set_volume(0.1)
        elif self.bttn_clicks1==2:
            mixer.init()
            mixer.music.set_volume(0.2)
        elif self.bttn_clicks1==3:
            mixer.init()
            mixer.music.set_volume(0.3)
        elif self.bttn_clicks1==4:
            mixer.init()
            mixer.music.set_volume(0.4)
        elif self.bttn_clicks1==5:
            mixer.init()
            mixer.music.set_volume(0.5) 
        elif self.bttn_clicks1==6:
            mixer.init()
            mixer.music.set_volume(0.6)
        elif self.bttn_clicks1==7:
            mixer.init()
            mixer.music.set_volume(0.7)
        elif self.bttn_clicks1==8:
            mixer.init()
            mixer.music.set_volume(0.8)
        elif self.bttn_clicks1==9:
            mixer.init()
            mixer.music.set_volume(0.9)
        elif self.bttn_clicks1==10:
            mixer.init()
            mixer.music.set_volume(1.0)
            
            self.bttn_clicks1=0 
        #print("Total Clicks: " + str(self.bttn_clicks))
        break

root = Tk()
app = MusicPlayer(root)
root.mainloop()
```
EDIT: I changed the volume if to use `get_volume` instead of `set_volume`

**Comment by Crilum** [comment]
There's actually quite a lot of functions in `pygame.mixer`: https://www.pygame.org/docs/ref/mixer.html

**Reply by Raspberry-Pi-Reviews** [reply]
yeah

**Reply by Raspberry-Pi-Reviews** [reply]
I got the same error

**Comment by Raspberry-Pi-Reviews** [comment]
@Crilum whats good my bro

**Reply by Crilum** [reply]
I'm actually learning Python, and PyQt5.

**Reply by Raspberry-Pi-Reviews** [reply]
oh shoot sorry I didnt know you messeged back, and thats cool.

**Reply by Raspberry-Pi-Reviews** [reply]
im gonna start working on a new gui design

**Reply by Crilum** [reply]
> oh shoot sorry I didnt know you messeged back, and thats cool.

It's fine.

**Reply by Crilum** [reply]
> im gonna start working on a new gui design

Nice! Still using Tkinter, I assume?

**Reply by Raspberry-Pi-Reviews** [reply]
Yeah, I tried to get the pyQt5 thing on my rpi 4 but I couldn't so im sticking to python

**Reply by Crilum** [reply]
> Yeah, I tried to get the pyQt5 thing on my rpi 4 but I couldn't so im sticking to python

It's not super hard:
Install PyQt5: `pip install PyQt5`
and then to create a mainWindow (a quick example I made):
```python
from PyQt5.QtWidgets import *


class MainWindow(QWidget):
    def __init__(self):
        super(MainWindow, self).__init__()
        self.setGeometry(100, 100, 500, 500)
        self.setWindowTitle("A Main window")
        label = QLabel(self)
        label.setText("some text")
        
        textBox = QTextEdit(self)
        textBox.setText("Editable text")
        button1 = QPushButton(self)
        button1.setText("A button")
        button2 = QPushButton(self)
        button2.setText("Another button")
        layout = QVBoxLayout()
        layout.addWidget(label)
        layout.addWidget(textBox)
        layout.addWidget(button1)
        layout.addWidget(button2)

        self.setLayout(layout)

        button1.clicked.connect(lambda: button1.setText("You clicked me!"))
        button2.clicked.connect(lambda: textBox.setText("You clicked button2!"))
        self.show()


def main():
    app = QApplication([])
    window = MainWindow()
    app.exec_()


if __name__ == '__main__':
    main()
```
You can also create a GUI by using Qt Designer, you just drag and drop the elements you want, and then you can power it with Python.
This is a good video to watch if you want to learn more: https://www.youtube.com/watch?v=MOItX2aKTGc

**Reply by Raspberry-Pi-Reviews** [reply]
okay sweet thanks bro!

**Reply by Raspberry-Pi-Reviews** [reply]
the code kind of looks a little like python, is it basically the same thing?

**Reply by Raspberry-Pi-Reviews** [reply]
that looks easier than python lol. what I think I was trying to do was the Qt designer and thats why it wasnt working

**Reply by Crilum** [reply]
> the code kind of looks a little like python, is it basically the same thing?

PyQt5 is a Python module, so you're just using Python, but with an outside library you imported. It's similar to how you use Tkinter. 

> that looks easier than python lol. what I think I was trying to do was the Qt designer and thats why it wasnt working

Yeah, you use Qt Designer to create the GUI, and then you program all the actual functionality with Python and PyQt5.
The video I linked in my last comment shows you how to use 100% Python and PyQt5, and also how to make the GUI with Qt Designer, and add the functionality with Python (and PyQt5).

**Reply by Raspberry-Pi-Reviews** [reply]
thanks, do you have like snap or something so I dont have to keep logging on here?

**Reply by Crilum** [reply]
You can email me at crilum@crilum.tk. I'll respond from a different address, just so you know. That address just forwards email to my personal email.

**Reply by Raspberry-Pi-Reviews** [reply]
oh thats cool how did you do that?

**Reply by Crilum** [reply]
I got a free domain name ([`crilum.tk`](https://ctilum.tk), which is a real website) from FreeNom.com, transfered the domain to Cloudflare's nameservers, and they have an Email forwarding service (on the free plan).

**Reply by Raspberry-Pi-Reviews** [reply]
oh nice

**Reply by Crilum** [reply]
Yeah, you can't send stuff from it, but it's great for keeping your email anonymous..

**Reply by Raspberry-Pi-Reviews** [reply]
sorry I havent been on in a while, my rpi has been at my dads and I kept forgeting to grab it but I didnt forgot it the last time so I have it now

**Reply by Crilum** [reply]
Nice. I actually don't have mine anymore..

**Reply by Raspberry-Pi-Reviews** [reply]
happy late new years lol

**Reply by Crilum** [reply]
Happy new year!!

**Reply by Raspberry-Pi-Reviews** [reply]
how you been bro?

**Reply by Crilum** [reply]
Good! I actually emailed you my Discord tag, so we tan talk there if ya like

**Reply by Raspberry-Pi-Reviews** [reply]
yooo you still active my boy

**Reply by Crilum** [reply]
Yooo wassup

Yes I am! kinda. I don't contribute much anymore, but I still like coding!

**Comment by Raspberry-Pi-Reviews** [comment]
Yeah same! Its been 5 years or so g, I was looking at the suda music again
cuz I got my raspberry pi back up and running, there's 2 versions that were
made, i don't know if you remember but I added like a playlist/track thing
on the right and then the song that was playing in the left, but it's good
talking to you again, we should start that back up it has a lot of
potential!
